exist-core/src/main/java/org/exist/xquery/functions/fn/FunHigherOrderFun.java

```diff
@@ -102,13 +104,14 @@ public FunHigherOrderFun(XQueryContext context, FunctionSignature signature) {
 	}
 
     @Override
-    protected void checkArguments() throws XPathException {
+	protected void checkArgument(final Expression arg, @Nullable final SequenceType argType,
+			final AnalyzeContextInfo argContextInfo, final int argPosition) throws XPathException {
         // hack: order of parameters for filter and other functions has changed
         // in final XQ3 spec. This would cause some core apps (dashboard) to stop
         // working. We thus switch parameters dynamically until all users can be expected to
         // have updated to 2.2.
         if (!isCalledAs("filter")) {
```

Comment:
Can this be removed?

Reply:
this was added in 2014 -- 0ecdf809530818ae1f8c74d5e7cb1ac80fa70d47
We should test, if this override can be removed entirely.